python/ray/train/tests/test_batch_predictor.py

```diff
@@ -154,6 +172,98 @@ def test_batch_prediction():
     ]
 
 
+def test_batch_prediction_various_combination():
+    """Dataset level predictor test against various
+    - Dataset format
+    - Preprocessor implementation (pandas vs numpy)
+    - Predictor implementation (pandas vs numpy)
+    """
+    # Got to inline this rather than using @pytest.mark.parametrize to void
+    # unknown object owner error when running test with python cli.
```

Comment:
Hmm that's odd. 🤔 I'm guessing that's due to the test cases involving `Dataset` objects? We can probably find a way to fix that, but I'm fine with doing that in a future PR.

Reply:
We do this other tests (like in `test_batch_mapper`)

Reply:
Yeah, agreed, this should work since it works as expected in `test_batch_mapper`.

Reply:
following up on this- we use parametrize in `test_batch_mapper`.

does that not work here?

Reply:
I fixed this issue for the `test_batch_mapper` tests by ensuring that the fixtures use the `ray_start_regular_shared` fixture for its Datasets execution, otherwise you could have the fixtures creating the Datasets on separate Ray clusters than the eventual tests run. If you have this test use the `ray_start_regular_shared` fixture, and turn these test cases into fixtures depending on the `ray_start_regular_shared` fixture, it should work: https://github.com/ray-project/ray/blob/c749ad380a722f2395dc49f4739ac84ddf87145e/python/ray/data/tests/conftest.py#L292

Can happen in a follow-up PR if you'd like.

Reply:
oh this is great to know, sure i can put up a follow up PR to make fixture work properly here